coil-core/src/androidInstrumentedTest/kotlin/coil3/decode/AndroidDecoderTest.kt

```diff
@@ -235,79 +268,75 @@ class BitmapFactoryDecoderTest {
+        // Android ImageDecoder handle exif internally so we cannot tune it
+        // Test BitmapFactoryDecoder only
+        assumeTrue(factory() is BitmapFactoryDecoder.Factory)
 
         // Test JPG
         for (index in 1..8) {
             val assetName = "exif/$index.jpg"
             val expected = BitmapFactory.decodeStream(context.assets.open(assetName))
-            val actual = decodeBitmap(assetName, Size.ORIGINAL, factory = factory)
+            val actual = decodeBitmap(assetName, Size.ORIGINAL)
             assertTrue(expected.isSimilarTo(actual), "Image with index $index is incorrect.")
         }
 
         // Test PNG
         for (index in 1..8) {
             val assetName = "exif/$index.png"
             val expected = BitmapFactory.decodeStream(context.assets.open(assetName))
-            val actual = decodeBitmap(assetName, Size.ORIGINAL, factory = factory)
+            val actual = decodeBitmap(assetName, Size.ORIGINAL)
             assertTrue(expected.isSimilarTo(actual), "Image with index $index is incorrect.")
         }
     }
 
     @Test
-    fun exifOrientationPolicy_respectPerformance() = runTest(timeout = 1.minutes) {
-        val factory = BitmapFactoryDecoder.Factory(
-            exifOrientationPolicy = ExifOrientationPolicy.RESPECT_PERFORMANCE,
-        )
-
+    fun exifOrientationPolicy_respectPerformance() = runTestPerDecoder(timeout = 1.minutes, exifOrientationPolicy = ExifOrientationPolicy.RESPECT_PERFORMANCE) {
         // Test JPG
-        val normalJpg = decodeBitmap("normal.jpg", Size.ORIGINAL, factory = factory)
+        val normalJpg = decodeBitmap("normal.jpg", Size.ORIGINAL)
         for (index in 1..8) {
-            val actual = decodeBitmap("exif/$index.jpg", Size.ORIGINAL, factory = factory)
+            val actual = decodeBitmap("exif/$index.jpg", Size.ORIGINAL)
             assertTrue(normalJpg.isSimilarTo(actual), "Image with index $index is incorrect.")
         }
 
         // Test PNG
         for (index in 1..8) {
             val assetName = "exif/$index.png"
             val expected = BitmapFactory.decodeStream(context.assets.open(assetName))
-            val actual = decodeBitmap(assetName, Size.ORIGINAL, factory = factory)
+            val actual = decodeBitmap(assetName, Size.ORIGINAL)
             assertTrue(expected.isSimilarTo(actual), "Image with index $index is incorrect.")
         }
     }
 
     @Test
-    fun exifOrientationPolicy_respectAll() = runTest(timeout = 1.minutes) {
-        val factory = BitmapFactoryDecoder.Factory(
-            exifOrientationPolicy = ExifOrientationPolicy.RESPECT_ALL,
-        )
+    fun exifOrientationPolicy_respectAll() = runTestPerDecoder(timeout = 1.minutes, exifOrientationPolicy = ExifOrientationPolicy.RESPECT_ALL) {
+        // Android ImageDecoder handle exif internally so we cannot tune it
+        // Test BitmapFactoryDecoder only
+        assumeTrue(factory() is BitmapFactoryDecoder.Factory)
```

Comment:
Same here. We need this test to verify `ImageDecoder` correctly account's for an image's EXIF data rotation.